scripts/byron-shelley/mkfiles.sh

```diff
@@ -0,0 +1,628 @@
+
+  # Stake addresses registration certs
+  cardano-cli shelley stake-address registration-certificate \
+      --stake-verification-key-file addresses/${ADDR}-stake.vkey \
+      --out-file addresses/${ADDR}-stake.reg.cert
+
+done
+
+# user N will delegate to pool N
+USER_POOL_N="1"
+
+for N in ${USER_POOL_N}; do
+
+  # Stake address delegation certs
+  cardano-cli shelley stake-address delegation-certificate \
+      --stake-verification-key-file addresses/user${N}-stake.vkey \
+      --cold-verification-key-file  node-pool${N}/shelley/operator.vkey \
+      --out-file addresses/user${N}-stake.deleg.cert
+
+  ln -s ../addresses/pool-owner${N}-stake.vkey node-pool${N}/owner.vkey
+  ln -s ../addresses/pool-owner${N}-stake.skey node-pool${N}/owner.skey
+
+done
+
+echo "Generated payment address keys, stake address keys,"
+echo "stake address regitration certs, and stake address delegatation certs"
+echo
+ls -1 addresses/
+echo "====================================================================="
+
+
+# Next is to make the stake pool registration cert
+
+for NODE in ${POOL_NODES}; do
+
+  cardano-cli shelley stake-pool registration-certificate \
+    --testnet-magic 42 \
+    --pool-pledge 0 --pool-cost 0 --pool-margin 0 \
+    --cold-verification-key-file             ${NODE}/shelley/operator.vkey \
+    --vrf-verification-key-file              ${NODE}/shelley/vrf.vkey \
+    --reward-account-verification-key-file   ${NODE}/owner.vkey \
+    --pool-owner-stake-verification-key-file ${NODE}/owner.vkey \
+    --out-file                               ${NODE}/registration.cert
+done
+
+echo "Generated stake pool registration certs:"
+ls -1 node-*/registration.cert
+echo "====================================================================="
+
+
+# Now we'll construct one whopper of a transaction that does everything
+# just to show off that we can, and to make the script shorter
+
+# We'll transfer all the funds to the user1, which delegates to pool1
+# We'll register certs to:
+#  1. register the pool-owner1 stake address
+#  2. register the stake pool 1
+#  3. register the user1 stake address
+#  4. delegate from the user1 stake address to the stake pool
+
+cardano-cli shelley transaction build-raw \
+    --ttl 1000 \
+    --fee 0 \
+    --tx-in $(cardano-cli shelley genesis initial-txin \
+                --testnet-magic 42 \
+                --verification-key-file shelley/utxo-keys/utxo1.vkey) \
+    --tx-out $(cat addresses/user1.addr)+${MAX_SUPPLY} \
+    --certificate-file addresses/pool-owner1-stake.reg.cert \
+    --certificate-file node-pool1/registration.cert \
+    --certificate-file addresses/user1-stake.reg.cert \
+    --certificate-file addresses/user1-stake.deleg.cert \
+    --out-file tx1.txbody
+
+
+# TODO: this will become the transaction to register the pool, etc. We'll need to pick the tx-in from the actual UTxO -- since it contains the txid, we'll have to query this via cardano-cli shelley query utxo.
```

Comment:
I think we do have all the necessary CLI functionality to handle this TODO now. I suggest leaving it in for now and merge, doing it in a follow-up PR.